### PR TITLE
Provide input to new workspaces topic

### PR DESCRIPTION
**Pull request by bpasero**

### CODE REVIEW

File: docs/editor/workspaces.md [hunk 1]
```diff
@@ -9,39 +9,91 @@ MetaDescription: Learn about Visual Studio Code workspaces
...
+**bpasero>** see my note in [1]
 Briefly describe `launch.json` and link to other docs TBD
 
 ### tasks
-
+**bpasero>** see my note in [1]
 Briefly describe `tasks.json` and link to other docs TBD
 
+[1] **bpasero>**
+I am trying to distill what we need to explain for settings, debugging and tasks in the context of explaining what a "workspace" is. My initial reaction is that all of this should have already been explained in the respective doc pages so I would not make this very detailed here to not repeat everything. However, I think one important concept is worthwhile to mention here: if you open a `.code-workspace` file (or actually when you are inside an "untitled" workspace), you get a third dimension of settings that you don't have otherwise:
```
Comment (gregvanl) [inline docs/editor/workspaces.md:55]: I think there is some value in mentioning the .vscode folder, which can include files such as settings.json, tasks.json, launch.json, extensions.json. Agreed that we don't need to go into too much detail.
Perhaps we can just have a bulleted list mentioning the files with a brief description of what they control with a link to the appropriate main doc.
Also have a quick mention that in a multi-root workspace there can be multiple .vscode folders contributing configurations.

File: docs/editor/workspaces.md [hunk 2]
```diff
@@ -9,39 +9,91 @@ MetaDescription: Learn about Visual Studio Code workspaces
...
+
+The same is true for multi-root workspaces that you can open easily by selecting the `<name>.code-workspace` file for opening.
+
+![Multi-folder workspace](images/workspaces/multi-folder-workspace.png)
 
+*A multi-root workspace opened in VS Code*
+
+**Note:** The visual difference of having a folder opened vs. opening a `.code-workspace` file can be very subtle. To give you a hint that a `.code-workspace` file has been opened, some pieces of the user interface (e.g. the root of the explorer) show an extra "(Workspace)" suffix next to the name.
+
+## Workspace state
```
Comment (gregvanl) [inline docs/editor/workspaces.md:34]: Agreed that we probably don't need this section or a detailed enumeration of persisted state.

File: docs/editor/workspaces.md [hunk 3]
```diff
@@ -9,39 +9,91 @@ MetaDescription: Learn about Visual Studio Code workspaces
...
+
+![Untitled multi-folder workspace](images/workspaces/untitled-workspace.png)
+
+*An untitled multi-root workspace opened in VS Code*
+
+There is really no difference between an untitled workspace and a saved workspace other than the fact that an untitled workspace is automatically created for you for your convinience and will always restore until you save it. We automatically delete untitled workspaces (after asking you for confirmation) when you close a window where the untitled workspace is opened in.
+
+**Note:** if you constantly see many windows restoring when you start VS Code, make sure to review the ones with untitled workspaces in them. To stop restoring untitled workspaces, simply close the window and confirm the prompt to not save the untitled workspace.
```
Comment (gregvanl) [inline docs/editor/workspaces.md:91]: How common is this behavior? Would it be better to have in Common Questions if few people hit this?

File: docs/editor/workspaces.md [hunk 4]
```diff
@@ -9,39 +9,91 @@ MetaDescription: Learn about Visual Studio Code workspaces
...
+		}
+	],
+    "settings": {
+        ...
+    }
+}
+```
+
+The presence of this `.code-workspace` file makes it even possible to include workspace specific settings (or launch / task configurations) that should apply to all folders of the workspace.
+
```
Comment (gregvanl) [inline docs/editor/workspaces.md:80]: We may also want to mention that in a multi-root workspace there can be multiple .vscode folders contributing configurations which are still scoped to the specific folders.

**Comment by bpasero** [comment]
@gregvanl thanks, I have made more changes. The doc is ready for a final review. Once you gave that I will probably also ask Sandeep from our team to have a look because he is closely involved with that topic too 👍

**Comment by gregvanl** [comment]
@bpasero Looks good. I'll do an edit pass with a follow up commit and then we can start sharing.

**Comment by bpasero** [comment]
@gregvanl awesome, thanks!

**Pull request merged**